cogs/helpers.py

```diff
@@ -563,6 +564,94 @@ async def colour(self, ctx, *, arg: str):
         fn = "{}.{}".format(match.group(1), ext)
         await ctx.send(file=discord.File(fp=buffer, filename=fn))
 
+    @commands.Cog.listener()
+    async def on_member_remove(self, member):
+        roles_id = []
+        for role in member.roles:
+            if role.name != "@everyone":
+                roles_id.append(role.id)
+        if len(roles_id) > 0:
+            conn = sqlite3.connect(self.bot.config.db_path)
+            c = conn.cursor()
+            t = (member.id, ' '.join(str(e) for e in roles_id))
+            c.execute('REPLACE INTO PreviousRoles VALUES (?, ?)', t)
+            conn.commit()
+            conn.close()
+
+    @commands.command(aliases=['previousroles', 'giverolesback', 'rolesback'])
+    async def previous_roles(self, ctx, user: discord.Member):
+        """Show the list of roles that a user had before leaving, if possible.
+        A moderator can click the OK react on the message to give these roles back
+        """
+        conn = sqlite3.connect(self.bot.config.db_path)
+        c = conn.cursor()
+        roles_id_string = c.execute(
+            'SELECT Roles FROM PreviousRoles WHERE ID = ?',
+            (user.id, )).fetchall()
```

Comment:
use fetchone instead of fetchall, roles_id_string is a confusing name because its not a string, its a tuple of one string

Reply:
Resolved in ed39449